ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Node.hs

```diff
@@ -1161,24 +1152,52 @@ protocolClientInfoCardano epochSlots = ProtocolClientInfo {
 -------------------------------------------------------------------------------}
 
 mkPartialLedgerConfigShelley ::
-     ShelleyGenesis (EraCrypto era)
-  -> Core.TranslationContext era
+     L.EraTransition era
+  => L.TransitionConfig era
   -> MaxMajorProtVer
   -> TriggerHardFork
   -> PartialLedgerConfig (ShelleyBlock proto era)
-mkPartialLedgerConfigShelley
-  genesisShelley
-  transCtxt
-  maxMajorProtVer
-  shelleyTriggerHardFork =
+mkPartialLedgerConfigShelley transitionConfig maxMajorProtVer shelleyTriggerHardFork =
     ShelleyPartialLedgerConfig {
           shelleyLedgerConfig =
             Shelley.mkShelleyLedgerConfig
-              genesisShelley
-              transCtxt
+              (transitionConfig ^. L.tcShelleyGenesisL)
+              (transitionConfig ^. L.tcTranslationContextL)
               -- 'completeLedgerConfig' will replace the 'History.dummyEpochInfo'
               -- in the partial ledger config with the correct one.
               History.dummyEpochInfo
               maxMajorProtVer
         , shelleyTriggerHardFork = shelleyTriggerHardFork
         }
+
+class
```

Comment:
Could these new classes use some comments?

Reply:
In my opinion this whole collection of classes is totally unnecessary, but I might be missing something.

Reply:
See the last paragraph in https://github.com/input-output-hk/ouroboros-consensus/pull/352#discussion_r1337399253

Will push a commit that shows how it looks like without these helper classes. I think we could eventually remove most per-era boilerplate in this file by adding even more such classes, but it actually seems more consistent to go the "pedestrial" route for now. :smile:

Reply:
See fdc687428b61a27bec27848e01c6e1b5bd38d270